Lib/test/test_shutil.py

```diff
@@ -1101,6 +1101,36 @@ def test_copytree_subdirectory(self):
         rv = shutil.copytree(src_dir, dst_dir)
         self.assertEqual(['pol'], os.listdir(rv))
 
+    def test_copytree_to_itself_gives_sensible_error_message(self):
+        base_dir = self.mkdtemp()
+        src_dir = os.path.join(base_dir, "src")
+        os.makedirs(src_dir)
+        create_file((src_dir, "somefilename"), "somecontent")
+        _assert_are_the_same_file_is_raised(src_dir, src_dir)
+
+    @os_helper.skip_unless_symlink
+    def test_copytree_to_backpointing_symlink_gives_sensible_error_message(self):
+        base_dir = self.mkdtemp()
+        src_dir = os.path.join(base_dir, "src")
+        target_dir = os.path.join(base_dir, "target")
+        os.makedirs(src_dir)
+        os.makedirs(target_dir)
+        some_file = os.path.join(src_dir, "somefilename")
+        create_file(some_file, "somecontent")
+        os.symlink(some_file, os.path.join(target_dir, "somefilename"))
+        _assert_are_the_same_file_is_raised(src_dir, target_dir)
+
+def _assert_are_the_same_file_is_raised(src_dir, target_dir):
+    try:
+        shutil.copytree(src_dir, target_dir, dirs_exist_ok=True)
+        raise Exception("copytree did not raise")
+    except Error as error:
+        assert len(error.args[0]) == 1
+        if sys.platform == "win32":
+            assert "it is being used by another process" in error.args[0][0][2]
+        else:
+            assert "are the same file" in error.args[0][0][2]
```

Comment:
We should be using `assertEqual` and `assertIn` via unittest instead of plain old assertions. Speaking of which, this is fine to be a method on the test class.

Reply:
fixed